### DISCUSSION TITLE
I can not Updating registry data

### DISCUSSION THREAD

**Discussion by Niezhanyang** [microsoft/vcpkg#38291] [Q&A]
Recently I am using the extended Arm KEil Pack to develop STM32
Among them I encountered a problem about vcpkg

I am using a 64-bit Ubuntu system: ubuntu 22.04.4

My console output is as follows
Updating registry data from https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip
failed to download from https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip
{}
failed to download /home/ubuntu/.vcpkg/downloads/bcee2a60-registry.zip from any source

My vcpkg-configuration.json file is as follows:
{
  "registries": [
   {
    "name": "arm",
    "kind": "artifact",
    "location": "https://artifacts.tools.arm.com/vcpkg-registry"
   }
  ],
  "requires": {
   "arm:tools/open-cmsis-pack/cmsis-toolbox": "2.3.0",
   "arm:tools/kitware/cmake": "3.28.4",
   "arm:tools/ninja-build/ninja": "1.11.1",
   "arm:compilers/arm/armclang": "6.22.0",
   "arm:debuggers/arm/armdbg": "6.1.1",
   "arm:models/arm/avh-fvp": "11.22.39",
   "arm:compilers/arm/arm-none-eabi-gcc": "13.2.1",
   "arm:compilers/arm/llvm-embedded": "17.0.1",
   "arm:tools/arm/uv2csolution": "1.3.0"
  }
}
My vcpkg-configuration.json file is as follows:
{
   "registries": [
     {
       "kind": "artifact",
       "name": "microsoft",
       "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip"
     },
     {
       "kind": "artifact",
       "name": "arm",
       "location": "https://artifacts.keil.arm.com/vcpkg-ce-registry/registry.zip"
     }
   ]
}

I guess it's a network problem, but I can download main.zip directly in the browser, so I rule out the network problem.
How should I solve this problem now?
I see someone has the same problem ,but it is not solved

**Comment by Niezhanyang** [comment]
I have solved this problem, I directly installed cmsis-toolbox manually and then it's works
this website will help https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/README.md
Since I live in an area with limited network access, I thought it was most likely a network issue.

**Comment by francovaro** [comment]
I have the same error with on the same environment. 
@Niezhanyang how did you solve it ? Because I have already installed cmsis-toolbox but I can't Build using the hammer icon and I get the error every time I open the folder.
I can build from terminal using cbuild.

**Comment by bpeddie** [comment]
I'm having the same problem on Ubuntu 24.02 - every time I open VSCode the following error occurs:

Updating registry data from https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip
failed to download from https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip
{}
failed to download /home/ubuntu/.vcpkg/downloads/bcee2a60-registry.zip from any source

I can't seem to find any reference to bcee2a60-registry.zip anywhere. Searches only point to this thread.

**Comment by djmorvay** [comment]
My issue occurred on a fresh install of Ubuntu and I did not have the appropriate dependencies installed. A simple 
sudo apt-get install curl zip unzip tar 
did the trick for me. It seems the installer couldn't unzip.

**Reply by bpeddie** [reply]
Thanks - this indeed fixed the issue for me on a 24.04 LTS install. Specifically -curl- was missing.

**Reply by dg0yt** [reply]
So the README is complete?
https://github.com/microsoft/vcpkg/blob/a4275b7eee79fb24ec2e135481ef5fce8b41c339/README.md?plain=1#L167-L177

**Reply by Mehdi00-7** [reply]
THanks

**Comment by ezlearning** [comment]
My issue is because arm version ubuntu, must set:
export VCPKG_FORCE_SYSTEM_BINARIES=1

~/.vscode/extensions/arm.environment-manager-1.12.0-linux-arm64/tools/vcpkg/vcpkg --help
Environment variable VCPKG_FORCE_SYSTEM_BINARIES must be set on arm, s390x, ppc64le and riscv platforms.
